core/trino-main/src/test/java/io/trino/execution/TestPageSplitterUtil.java

```diff
@@ -95,8 +98,89 @@ public void testSplitPageNonDecreasingPageSize()
         assertThat((int) first.getSizeInBytes()).isGreaterThan(maxPageSizeInBytes);
         assertThat((int) second.getSizeInBytes()).isGreaterThan(maxPageSizeInBytes);
         assertPositionCount(pages, positionCount);
-        MaterializedResult actual = toMaterializedResult(TEST_SESSION, types, pages);
-        MaterializedResult expected = toMaterializedResult(TEST_SESSION, types, ImmutableList.of(initialPage));
-        assertThat(actual).containsExactlyElementsOf(expected);
+    }
+
+    // Fake block that has retains a fixed size when split
+    private record FixedDataSizeBlock(long fixedSize, ValueBlock delegate)
```

Comment:
This class seems like it's just mimicking the old behavior of RLE blocks, which no longer is the case. I think having to change the test assertions here to use a new fake block type is masking the real problem, which is that `PageSplitterUtil` will now split RLE blocks very aggressively and unnecessarily into small blocks that will hurt performance.

If we keep the change to RLE blocks, then there's a large block comment inside of `PageSplitterUtil#splitPage` describing (now incorrectly) how RLE blocks don't change size when split that you'll want to remove.

Reply:
I've updated the code comment in PageSplitterUtil and added a commit to increase the threshold for splitting pages to 2MB. I think the existing limit was already low considering that the limit for http request size (for serialized page) is much higher and the limit on size of pages produced by parquet and orc readers is also 8MB.